frontend/src/pages/secret-manager/OverviewPage/components/SecretOverviewTableRow/SecretRenameRow.tsx

```diff
@@ -196,8 +196,12 @@ function SecretRenameRow({ environments, getSecretByKey, secretKey, secretPath }
         />
       </div>
 
-      {isReadOnly || isOverriden ? (
-        <span className="mr-5 rounded-md bg-mineshaft-500 px-2">Read Only</span>
+      {isReadOnly || isOverridden ? (
+        <Tooltip
+          content={isReadOnly ? "Not allowed" : "Remove personal overrides to change the key"}
+        >
+          <span className="mr-5 rounded-md bg-mineshaft-500 px-2">Read Only</span>
+        </Tooltip>
```

Comment:
While using the overview, it wasn't clear why key was made read only in some cases.